automations/js/src/count_user_reviewable_prs.js

```diff
@@ -38,6 +38,7 @@ query ($repoOwner: String!, $repo: String!, $cursor: String) {
           }
         }
         isDraft
+        createdAt
```

Comment:
The event that triggers this script already has the PR in context (https://docs.github.com/en/webhooks/webhook-events-and-payloads#pull_request) from which we can retrieve the latest PR. @dhruvkb told me GitHub's rate limits for GraphQL are a bit different (https://docs.github.com/en/graphql/overview/rate-limits-and-node-limits-for-the-graphql-api#primary-rate-limit) so anything we can avoid requesting would be good to avoid.

Instead of getting the most recent PR this way, we can pass the `pull_request` context object to the script and check if it's a relevant PR at the top, and avoid making any requests at all if it isn't. Something like:

```
def (..., pullRequest):
  if not isRelevantPR(pullRequest):
    return { pr_count: 0 }

  # the rest of the function
```

Reply:
It looks like we have a few helper methods for getting the pull request object in other scripts:

https://github.com/WordPress/openverse/blob/8b0ec498e12d73a0fc3499a5a8f5753911b2fe9c/automations/js/src/label_pr.mjs#L57-L72

https://github.com/WordPress/openverse/blob/8cd1dbcf8861238bac15dcaae28b587f244a377c/automations/js/src/project_automation/prs.mjs#L67-L75

These use GraphQL under the hood themselves, so I'm not sure we'd change cut down much on usage by using this method instead. Is it really as simple as putting `pull_request` in the exports here? https://github.com/WordPress/openverse/blob/b8230bea5c9ca00a14b69c0622f7862fd046fadb/automations/js/src/count_user_reviewable_prs.js#L11

Do you know where I might be able to find the shape of the object that's available when I do? I'm struggling with the documentation for Octokit 😅

Reply:
No, you'd need to get it in from the GitHub action context. You _might_ be able to get it from `context.payload.pull_request`, and that _might_ have the same shape as the `github.event.pull_request` present in the YAML contexts, [which I found a dump of via this SO answer](https://stackoverflow.com/questions/70104600/complete-list-of-github-actions-contexts):

https://gist.github.com/colbyfayock/1710edb9f47ceda0569844f791403e7e#file-github-context-json-L19

So something like `context.payload.pull_request.created_at` could get you the creation date.

Those other scripts you found are pulling the PR information out of a temporary file built from the event context, and is only necessary because of what Dhruv explains in this comment: https://github.com/WordPress/openverse/blob/889c3b5959a26c18cf09b8be652b7992640f527d/.github/workflows/pr_automations_init.yml#L1-L18

It isn't relevant for this particular script because this script only runs on `pull_request_target` and no other event type, and so will always have the `pull_request` in the event context (unlike `pull_request_review`).

If for some reason `context.payload.pull_request` is _not_ a full `pull_request` object, then you can add it as a property on the ` module.exports = async ({ github, context, core }) => { ` you shared as:

```js
 module.exports = async ({ github, context, core, triggeringPullRequest }) => { 
```

and then pass that in the workflow file like so:

https://github.com/WordPress/openverse/blob/889c3b5959a26c18cf09b8be652b7992640f527d/.github/workflows/pr_limit_reminders.yml#L24

to

```yml
await script({ github, context, core, triggeringPullRequest: ${{ toJSON(github.event.pull_request) })
```

I'm hoping you don't need to do that though and that we can get it from `context.payload.pull_request`. I'm sharing this alternative _just in case_ the simpler approach doesn't work. Unfortunately GitHub Action contexts are not super well documented (as you can see from that SO question I linked).

Reply:
Thanks for all that info! And amazing find with that gist. I think I've got a workable approach in dfa97590327e9da9c06a20cb00a0e941a01629bd - I had to add a separate function for checking relevancy since the shape of the context's PR is different from the one returned by the GraphQL API. Let me know what you think!